modules/core/src/webgl/classes/texture-formats.js

```diff
@@ -11,50 +11,153 @@ import {isWebGL2} from '../utils';
   // we need to update this table structure to specify extensions (gl1: 'OES_texure_float', gl2: false) for each type.
   [GL.RGBA]: {
     dataFormat: GL.RGBA,
-    types: [GL.UNSIGNED_BYTE, GL.UNSIGNED_SHORT_4_4_4_4, GL.UNSIGNED_SHORT_5_5_5_1]
+    types: [GL.UNSIGNED_BYTE, GL.UNSIGNED_SHORT_4_4_4_4, GL.UNSIGNED_SHORT_5_5_5_1],
+    channels: 4
   },
-  [GL.ALPHA]: {dataFormat: GL.ALPHA, types: [GL.UNSIGNED_BYTE]},
-  [GL.LUMINANCE]: {dataFormat: GL.LUMINANCE, types: [GL.UNSIGNED_BYTE]},
-  [GL.LUMINANCE_ALPHA]: {dataFormat: GL.LUMINANCE_ALPHA, types: [GL.UNSIGNED_BYTE]},
+  [GL.ALPHA]: {dataFormat: GL.ALPHA, types: [GL.UNSIGNED_BYTE], channels: 1},
+  [GL.LUMINANCE]: {dataFormat: GL.LUMINANCE, types: [GL.UNSIGNED_BYTE], channels: 1},
+  [GL.LUMINANCE_ALPHA]: {dataFormat: GL.LUMINANCE_ALPHA, types: [GL.UNSIGNED_BYTE], channels: 2},
 
   // 32 bit floats
-  [GL.R32F]: {dataFormat: GL.RED, types: [GL.FLOAT], gl2: true},
-  [GL.RG32F]: {dataFormat: GL.RG, types: [GL.FLOAT], gl2: true},
-  [GL.RGB32F]: {dataFormat: GL.RGB, types: [GL.FLOAT], gl2: true},
-  [GL.RGBA32F]: {dataFormat: GL.RGBA, types: [GL.FLOAT], gl2: true}
-
-  // [GL.DEPTH_COMPONENT]: {types: [GL.UNSIGNED_SHORT, GL.UNSIGNED_INT, GL.UNSIGNED_INT_24_8], gl1: DEPTH},
-  // [GL.DEPTH_STENCIL]: {gl1: DEPTH},
+  [GL.R32F]: {dataFormat: GL.RED, types: [GL.FLOAT], gl2: true, bytesPerTexel: 4},
+  [GL.RG32F]: {dataFormat: GL.RG, types: [GL.FLOAT], gl2: true, bytesPerTexel: 8},
+  [GL.RGB32F]: {dataFormat: GL.RGB, types: [GL.FLOAT], gl2: true, bytesPerTexel: 12},
+  [GL.RGBA32F]: {dataFormat: GL.RGBA, types: [GL.FLOAT], gl2: true, bytesPerTexel: 16},
+
+  // Depth/Stencil
+  [GL.DEPTH_COMPONENT]: {
+    dataFormat: GL.DEPTH_COMPONENT,
+    types: [GL.UNSIGNED_SHORT, GL.UNSIGNED_INT],
+    gl2: false,
+    gl1: 'WEBGL_depth_texture',
+    channels: 1
+  },
+  [GL.DEPTH_COMPONENT16]: {
+    dataFormat: GL.DEPTH_COMPONENT,
+    types: [GL.UNSIGNED_SHORT, GL.UNSIGNED_INT],
+    gl2: true,
+    bytesPerTexel: 2
+  },
+  [GL.DEPTH_COMPONENT24]: {
+    dataFormat: GL.DEPTH_COMPONENT,
+    types: [GL.UNSIGNED_INT],
+    gl2: true,
+    bytesPerTexel: 3
+  },
+  [GL.DEPTH_COMPONENT32F]: {
+    dataFormat: GL.DEPTH_COMPONENT,
+    types: [GL.FLOAT],
+    gl2: true,
+    bytesPerTexel: 4
+  },
+  [GL.DEPTH_STENCIL]: {
+    dataFormat: GL.DEPTH_STENCIL,
+    types: [GL.UNSIGNED_INT_24_8],
+    gl2: false,
+    gl1: 'WEBGL_depth_texture',
+    bytesPerTexel: 4
+  },
+  [GL.DEPTH24_STENCIL8]: {
+    dataFormat: GL.DEPTH_STENCIL,
+    types: [GL.UNSIGNED_INT_24_8],
+    gl2: true,
+    bytesPerTexel: 4
+  },
+  [GL.DEPTH32F_STENCIL8]: {
+    dataFormat: GL.DEPTH_STENCIL,
+    types: [GL.FLOAT_32_UNSIGNED_INT_24_8_REV],
+    gl2: true,
+    bytesPerTexel: 5
+  },
 
-  // Sized texture format - more performance
+  // Sized/compressed
   // R
-  // [GL.R8]: {dataFormat: GL.RED, types: [GL.UNSIGNED_BYTE], gl2: true},
```

Comment:
So this was commented out for bundle size reasons. Would be good if you could run collect metrics before and after this change. And make a pass on making the table smaller. That could be a second PR, no need to hold this one, but let's not just add this without checking.

Reply:
Ok, sure. I'm thinking what I'll do is create another table `DATA_FORMAT_CHANNELS`, which will be quite small, and then I can do the calculations for most cases based on `numChannels * typeSize`.

Might not be an issue for our current use cases, but it seems odd that we're returning "not supported" for a lot of useful WebGL 2 formats